src/rational/q/from.rs

```diff
@@ -249,6 +276,26 @@ impl<T: Into<Z> + Clone> From<&T> for Q {
     }
 }
 
+impl From<f64> for Q {
```

Comment:
You could also use the `from` macro here and also implement it for f32, f16, and f8 s.t. we do not have to touch it again.